diskpy/ICgen/pos_class.py

```diff
@@ -117,6 +117,67 @@ def _generate_r(self):
         print 'Generating r positions'
         cdf_inv_r = self._parent.sigma.cdf_inv
         
+        if self.method == 'glass':
+            sn = pynbody.load("/home/wadsley/python/glass16.std")
+            #Need to get a cylinder radius 1 and from z=-1 to +1 with self.nParticles in it
+            # Careful density in cylinder not uniform -- increases with radius (fix is to say r=x**2+y**2)
+            #Note: final distribution very flat -- ideally need to know something about h/r ratio
+            #Make a cylinder a little bigger than needed, 
+            #get r of particles, sort and then keep nParticles with smallest r's 
+            #reset last few to be inside if needed
+            #calculate thetas
+            nParticles = self.nParticles
+            hratio = 0.05
+            rscale = 1./(nParticles/(2*np.pi*4096)/hratio)**.3333333333333333
+            zscale = rscale/hratio
```

Comment:
zscale can be calculated as a function of R, but we need to be careful.  What we want is a cube in our first space to map as uniformly as possible to a cube in our final space (the actual ICs).  This will in general be a function of the surface density compared to the vertical density profile.